hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/CloseContainerEventHandler.java

```diff
@@ -104,14 +113,19 @@ public void onMessage(ContainerID containerID, EventPublisher publisher) {
         command.setTerm(scmContext.getTermOfLeader());
         command.setEncodedToken(getContainerToken(containerID));
 
-        getNodes(container).forEach(node ->
-            publisher.fireEvent(DATANODE_COMMAND,
-                new CommandForDatanode<>(node.getUuid(), command)));
+        try {
+          leaseManager.acquire(command, timeout, () -> triggerCloseCallback(
+              publisher, container, command));
```

Comment:
Initially I thought we could do
```
leaseManager.acquire(command, timeout).registerCallBack(() -> triggerCloseCallback(
              publisher, container, command));
```
but this has a subtle race since `acquire` may have already notified listeners by the time the callback is added. We don't have to change it here, but in a follow up PR it might be good to change the lease manager so this isn't possible, maybe by not returning the `Lease` object from `acquire`. We could also remove dead code in `EventWatcher` that is doing this currently.

Reply:
@errose28 Current usages has no problem, and to remove unused code which is having unsafe usages of lease manager method, [HDDS-8418](https://issues.apache.org/jira/browse/HDDS-8418) is added.